### PR TITLE
Add additional temperature debugging for RRTMGP

### PR DESCRIPTION
**Pull request by brhillman**
Add check to make sure temperatures are within bounds of absorption coefficient look-up tables at a higher level in the RRTMGP call tree in order to provide information about where temperatures go out of bounds. Additionally, add an option (specified by namelist variable `rrtmgp_clip_temperatures`) to clip temperatures when they go out of bounds. With `rrtmgp_clip_temperatures = .false.` (the default), the model will abort when temperatures out of bounds are encountered. With `rrtmgp_clip_temperatures = .true.`, the model will issue a warning that temperatures have gone out of bounds, but clip the temperature input to RRTMGP and continue.

### CODE REVIEW

File: components/cam/src/physics/rrtmgp/radiation.F90 [hunk 1]
```diff
@@ -124,7 +124,11 @@ module radiation
 
    ! k-distribution coefficients files to read from. These are set via namelist
    ! variables.
-   character(len=cl) :: coefficients_file_sw, coefficients_file_lw
+   character(len=cl) :: rrtmgp_coefficients_file_sw, rrtmgp_coefficients_file_lw
```
Comment (AaronDonahue) [inline components/cam/src/physics/rrtmgp/radiation.F90]: Should the comment above be changed to reflect this change in variable name?

File: components/cam/src/physics/rrtmgp/radiation.F90 [hunk 2]
```diff
@@ -1373,6 +1382,25 @@ subroutine radiation_driver_sw(state, pbuf, cam_in, is_cmip6_volc, &
                          pint(1:nday,1:nlev_rad+1), &
                          col_indices=day_indices(1:nday))
 
+      ! Check temperatures to make sure they are within the bounds of the
+      ! absorption coefficient look-up tables
+      do iday = 1,nday
+         icol = day_indices(iday)
+         lat(iday) = state%lat(icol) * pi / 180._r8
```
Comment (AaronDonahue) [inline components/cam/src/physics/rrtmgp/radiation.F90]: Have you checked this?  If state%lat is in radians isn't the conversion ( *180 / pi )

File: components/cam/src/physics/rrtmgp/radiation.F90 [hunk 3]
```diff
@@ -1561,7 +1594,26 @@ subroutine radiation_driver_lw(state, pbuf, cam_in, is_cmip6_volc, &
...
                          pmid(1:ncol,1:nlev_rad), &
                          pint(1:ncol,1:nlev_rad+1))
-       
+
+      ! Check temperatures to make sure they are within the bounds of the
+      ! absorption coefficient look-up tables. If out of bounds, optionally clip
+      ! values to min/max specified (depending on value of
+      ! rrtmgp_clip_temperatures)
+      do icol = 1,ncol
+         lat(icol) = state%lat(icol) * pi / 180._r8
```
Comment (AaronDonahue) [inline components/cam/src/physics/rrtmgp/radiation.F90]: See comment above

File: components/cam/src/physics/rrtmgp/radiation.F90 (hunk 2 above)
Comment (brhillman) [reply components/cam/src/physics/rrtmgp/radiation.F90]: Well duh, of course it is. Will push fix.

**Comment by AaronDonahue** [comment]
@brhillman , should we go ahead and switch this to "AUTOMERGE" since we want to get it in ASAP?

**Pull request merged**